internal/antewrapper/fee_gas_meter.go

```diff
@@ -112,13 +112,13 @@ func (g *FeeGasMeter) String() string {
 }
 
 // ConsumeFee increments the amount of msg fee required by a msg type.
-func (g *FeeGasMeter) ConsumeFee(amount sdk.Coin, msgType string, recipient string) {
+func (g *FeeGasMeter) ConsumeFee(amount sdk.Coins, msgType string, recipient string) {
```

Comment:
I'm sure this is fine after reviewing the code, but it assumes a msg fee with a msgType can have multiple coins associated with it.  Which is not the case. I know the internal property uses coins instead of coin in the FeeGasMeter.  `usedFees:       make(map[string]sdk.Coins),` maybe that should be changed to coin.  That way we would get a panic if there is a mix of coin denoms being added.

Reply:
pretty sure there was a test above that uses stake/hotdog together and works ... but yeah, I don't see a situation where we would have more than one denom although now with the ability to put 100% of the fee into an account other than the fee distribution module account I suppose it could be done.  The previous issue was seeing that alternate denom distributed to delegators and potential issues with that.

Reply:
Do we really care if there's multiple denom types in here though? Since all the downstream stuff handles having multiple denoms just fine, that seems like an unneeded limitation. I get that we only want a single denom for the msg fee because defining splitting for multiple denoms is burdensome, but I see no reason to care about it at this point.

Plus, changing the `amount` to `Coins` allowed me to remove some duplicated logic and just use the results of `CalculateAdditionalFeesToBePaid`.

Reply:
Since `fees` are `coins` everywhere in the sdk there is precedent here for using the coins value... we would only need to ensure we have solid unit tests/errors either way in my opinion.

Reply:
I'm fine with leaving it then.  👍